libp2p/peerstore.nim

```diff
@@ -70,11 +75,15 @@ type
 
   PeerStore* {.public.} = ref object
     books: Table[string, BasePeerBook]
+    identify: Identify
```

Comment:
Should the PeerStore have a protocol?

Reply:
"who is responsible for Identify" is a good question, previously this was a mix of the dialer, upgrader & connMngr. With this PR, this unifies it to "just the PeerStore", but we can change it to something else later

Reply:
I was thinking about `PeerInfo`, but it isn't in the `dialer`. How about having it in the `dialer` itself?